File: client/src/app/_models/mission-filter.ts
export interface MissionFilter {
  name?: string;
  status?: MissionStatus;
  /** ถ้าระบุ จะกรองภารกิจที่ไม่ใช่ของตัวเอง และยังไม่ได้เข้าร่วม */
  exclude_brawler_id?: number;
}

export type MissionStatus = 'Open' | 'InProgress' | 'Completed' | 'Failed' | '';

File: server/src/domain/value_objects/mission_filter.rs
use serde::{Deserialize, Serialize};

use crate::domain::value_objects::mission_statuses::MissionStatuses;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq, Default)]
pub struct MissionFilter {
    pub name: Option<String>,
    pub status: Option<MissionStatuses>,
    /// ถ้าระบุค่านี้ จะกรองภารกิจที่:
    /// 1. ไม่ใช่ของ brawler คนนี้ (chief_id ≠ exclude_brawler_id)
    /// 2. brawler คนนี้ยังไม่ได้เข้าร่วม (ไม่อยู่ใน crew_memberships)
    pub exclude_brawler_id: Option<i32>,
}
